pkg/migration/generate.go

```diff
@@ -1152,6 +1158,17 @@ func initRuntimeOptions(cluster *corev1.StorageCluster) {
 	}
 }
 
+func getStorageClusterNameFromClusterID(clusterID string) string {
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
+	// If the cluster ID is a valid k8s object name, just set it as the storage cluster name,
+	// otherwise format it to be k8s compatible
+	validNameRegex := regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`)
```

Comment:
we don't need to check here, why not directly use pxutil.GetK8sCompatibleObjectName?

Reply:
my understand is, if the original cluster ID passed to the px container args is valid for an object name then just use it, so we don't need to set the annotation. the util function here is more strict and trims more chars from the name, so don't use it unless necessary. if we call the util function every time, we need to set the annotation always.

Reply:
GetK8sCompatibleObjectName should not remove more chars than needed. above regex can be put in GetK8sCompatibleObjectName for validation and replacement (if the regex does not work for replacing entire string, we can still validate each char and remove)

if we call the util function every time, we need to set the annotation always.
you only add the annotation when the object is not the same as -c arg, that's after the util function is called anyway. correct?

Reply:
@haibinxie, he is using a much relaxed regex for validation, which will avoid unnecessary name replacement. 
`GetK8sCompatibleObjectName` function removes more chars because it is used by the configmap names. Unfortunately, that is how we will have to keep it to be consistent with the px generated names.